File: brawl-core/src/main/java/brawl/example/project_brawl_api_sheets/integration_sheets/config/SheetsConfig.java
package brawl.example.project_brawl_api_sheets.integration_sheets.config;

import com.google.api.client.googleapis.auth.oauth2.GoogleCredential;
import com.google.api.client.googleapis.javanet.GoogleNetHttpTransport;
import com.google.api.client.json.JsonFactory;
import com.google.api.client.json.jackson2.JacksonFactory;
import com.google.api.services.sheets.v4.Sheets;
import com.google.api.services.sheets.v4.SheetsScopes;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

import java.io.FileInputStream;
import java.io.IOException;
import java.security.GeneralSecurityException;
import java.util.Collections;
@Configuration
public class SheetsConfig {
    private static final String APPLICATION_NAME = "Planilha do Capeta";
    private static final JsonFactory JSON_FACTORY = JacksonFactory.getDefaultInstance();

    @Value("${credential.json}")
    private String CREDENTIALS_FILE_PATH;

    @Bean
    public Sheets sheetsService() throws IOException, GeneralSecurityException {
        GoogleCredential credential = GoogleCredential.fromStream(new FileInputStream(CREDENTIALS_FILE_PATH))
                .createScoped(Collections.singleton(SheetsScopes.SPREADSHEETS));

        return new Sheets.Builder(GoogleNetHttpTransport.newTrustedTransport(), JSON_FACTORY, credential)
                .setApplicationName(APPLICATION_NAME)
                .build();
    }
}

File: data-analysis/src/main/kotlin/ml/config/SheetsConfig.kt
package com.guilherme.ml.config

import com.google.api.client.googleapis.auth.oauth2.GoogleCredential
import com.google.api.client.googleapis.javanet.GoogleNetHttpTransport
import com.google.api.client.json.jackson2.JacksonFactory
import com.google.api.services.sheets.v4.Sheets
import com.google.api.services.sheets.v4.SheetsScopes
import org.springframework.context.annotation.Configuration
import java.io.FileInputStream

@Configuration
class SheetsConfig {
     var spreadsheetId = "SEU_ID_DA_PLANILHA_AQUI"
    private val APPLICATION_NAME = "Brawl Stars Sheets"
    private val JSON_FACTORY = JacksonFactory.getDefaultInstance()
    private val SCOPES = listOf(SheetsScopes.SPREADSHEETS_READONLY)

    //Por enquanto Indefinido
    private val CREDENTIALS_FILE_PATH = "credentials.json";

    fun getSheets() : Sheets {
        val httpTransport = GoogleNetHttpTransport.newTrustedTransport()

        val credentials = GoogleCredential.fromStream((FileInputStream(CREDENTIALS_FILE_PATH)))
            .createScoped(SCOPES);

        return Sheets.Builder(httpTransport, JSON_FACTORY, credentials)
            .setApplicationName(APPLICATION_NAME)
            .build();
    }

}